### PR TITLE
Add GeneralMenuBlock for better control of secondary navigation in desktop

### PR DESCRIPTION
**Pull request by gustaveen**
## Describe your changes
- Add new block GeneralMenu that holds the about Hedvig menu to better control behaviour
- In desktop, the menu item label is replaced with a hamburger icon and is triggered on click. This is alos the right behaviour on mobile, today it's very buggy since you can't open the sub menu in mobile
- Move HamburgerIcon to it's own file

https://github.com/HedvigInsurance/racoon/assets/6661511/c135a0e7-5148-4618-b2a2-b8e3a042bd46

Next steps:
- Fix submenu in mobile menu to toggle child items and not redirect

## Checklist before requesting a review

- [ ] I have performed a self-review of my code

### CODE REVIEW

**Comment by gustaveen** [comment]
* **#4460** <a href="https://app.graphite.dev/github/pr/HedvigInsurance/racoon/4460?utm_source=stack-comment-icon" target="_blank">Graphite</a> 👈
* `main`

This stack of pull requests is managed by Graphite. <a href="https://stacking.dev/?utm_source=stack-comment">Learn more about stacking.</a>
<h2></h2>

 Join @gustaveen and the rest of your teammates on <a href="https://graphite.dev?utm-source=stack-comment">Graphite <b>Graphite</b></a>

File: apps/store/src/components/HeaderNew/NavigationTriggerGeneralMenu.tsx [hunk 1]
```diff
@@ -0,0 +1,28 @@
...
+export const NavigationTriggerGeneralMenu = ({ children }: { children: string }) => {
+  return (
+    <NavigationMenuPrimitive.Trigger
+      className={navigationTriggerLink}
+      style={assignInlineVars({
+        [navigationTriggerLinkPadding]: '0',
+      })}
+      onPointerEnter={(event) => event.preventDefault()}
+      onPointerLeave={(event) => event.preventDefault()}
+      onPointerMove={(event) => event.preventDefault()}
```
Comment (gustaveen) [inline apps/store/src/components/HeaderNew/NavigationTriggerGeneralMenu.tsx:21]: Prevent all hover triggers

File: apps/store/src/blocks/HeaderBlockNew/GeneralMenuBlock.tsx [hunk 2]
```diff
@@ -0,0 +1,63 @@
...
+  }
+
+  return (
+    <NavigationMenuPrimitive.Item
+      className={clsx(navigationItem, navigationItemGeneralMenu)}
+      value={blok.name}
+      {...storyblokEditable(blok)}
+    >
+      <NavigationTriggerGeneralMenu>{blok.name}</NavigationTriggerGeneralMenu>
+      <NavigationContent onPointerLeave={(event) => event.preventDefault()}>
```
Comment (gustaveen) [inline apps/store/src/blocks/HeaderBlockNew/GeneralMenuBlock.tsx:38]: Prevent closing menu when cursor leaves content

Comment (alebedev) [reply apps/store/src/blocks/HeaderBlockNew/GeneralMenuBlock.tsx:38]: Can you comment this in the code?

File: apps/store/src/components/HeaderNew/NavigationTriggerGeneralMenu.tsx (hunk 1 above)
Comment (alebedev) [reply apps/store/src/components/HeaderNew/NavigationTriggerGeneralMenu.tsx:21]: Same, would be nice to have this comment inline

**Review by alebedev — APPROVED** [review]

**Pull request merged**